wurst/systems/spawns/ThiefBushWaterfall.wurst

```diff
@@ -1,24 +1,29 @@
 package ThiefBushWaterfall
 import HashMap
+import JumpSystem
+import TerrainUtils
+import ErrorHandling
 
 @configurable let THIEF_BUSH_WATERFALL_HEIGHT = 140.0
 @configurable let THIEF_BUSH_WATERFALL_DURATION = 1.0
 @configurable let THIEF_BUSH_WATERFALL_GRAVITY = 50.0
 
 let entranceToExitMap = new HashMap<region, rect>()
 
-function waterfallEntrance() returns boolean
+function waterfallEntrance()
     let hero = GetEnteringUnit()
     if hero.isType(UNIT_TYPE_GROUND)
         let exit = entranceToExitMap.get(GetTriggeringRegion())
-        let dest = exit.randomPoint()
-        // hero.pause()
-        // s__Jump_start(hero, dest.x, dest.y, THIEF_BUSH_WATERFALL_HEIGHT, THIEF_BUSH_WATERFALL_DURATION, THIEF_BUSH_WATERFALL_GRAVITY)
-    return false
-
-function waterfallExit() returns boolean
-    // EVENT_JUMP_UNIT.unpause()
-    return false
+        var dest = exit.randomPoint()

```

Comment:
This still has the issue where the dest may not be walkable. I'd try something along the lines like:

```
for i = 0 to MAX_TRIES
    var dest = exit.randomPoint()
    if dest.isTerrainWalkable()
        thiefJump(hero, dest)
        return
// Failed to find target point.
error("Could not find a suitable target to jump to.")
```